pallets/block-rewards/src/lib.rs

```diff
@@ -0,0 +1,440 @@
+		/// the same id.
+		#[pallet::constant]
+		type MaxChangesPerEpoch: Get<u32> + TypeInfo + sp_std::fmt::Debug + Clone + PartialEq;
+
+		#[pallet::constant]
+		type CollatorCurrencyId: Get<Self::CurrencyId> + TypeInfo;
+
+		#[pallet::constant]
+		type CollatorGroupId: Get<Self::GroupId> + TypeInfo;
+
+		#[pallet::constant]
+		type DefaultCollatorStake: Get<Self::Balance> + TypeInfo;
+
+		#[pallet::constant]
+		type MaxCollators: Get<u32> + TypeInfo + sp_std::fmt::Debug + Clone + PartialEq;
+
+		/// Identifier for the currency used to give the reward.
+		type RewardCurrency: Get<Self::CurrencyId>;
+
+		#[pallet::constant]
+		type RemainingRewardCollector: Get<Option<PalletId>>;
+
+		/// The identifier type for an authority.
+		type AuthorityId: Member
+			+ Parameter
+			+ sp_runtime::RuntimeAppPublic
+			+ MaybeSerializeDeserialize
+			+ MaxEncodedLen;
+
+		/// Information of runtime weights
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// Data associated to the current epoch.
+	#[pallet::storage]
+	pub(super) type ActiveEpochData<T: Config> = StorageValue<_, EpochData<T>, ValueQuery>;
+
+	/// Pending update data used when the current epoch finalizes.
+	/// Once it's used for the update, it's reset.
+	#[pallet::storage]
+	pub(super) type NextEpochChanges<T: Config> = StorageValue<_, EpochChanges<T>, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NewEpoch {
+			reward: T::Balance,
+			last_changes: EpochChanges<T>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Limit of max calls with same id to [`Pallet::set_group_weight()`] or
+		/// [`Pallet::set_currency_group()`] reached.
+		MaxChangesPerEpochReached,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Claims the reward the associated to a currency.
+		/// The reward will be transferred to the target account.
+		#[pallet::weight(T::WeightInfo::claim_reward())]
+		#[transactional]
+		pub fn claim_reward(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			account_id: T::AccountId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			T::Rewards::claim_reward((T::Domain::get(), currency_id), &account_id).map(|_| ())
+		}
+
+		/// Admin method to set the reward amount used for the next epochs.
+		/// Current epoch is not affected by this call.
+		#[pallet::weight(T::WeightInfo::set_distributed_reward())]
+		pub fn set_distributed_reward(origin: OriginFor<T>, balance: T::Balance) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			NextEpochChanges::<T>::mutate(|changes| changes.reward = Some(balance));
+
+			Ok(())
+		}
+
+		/// Admin method to set the group weights used for the next epochs.
+		/// Current epoch is not affected by this call.
+		#[pallet::weight(T::WeightInfo::set_group_weight())]
+		pub fn set_group_weight(
```

Comment:
I thought we finally decided to have only one group, but I probably remember badly. What's the use case for several groups?

Reply:
Yes, we decided on one group. I haven't abstracted that away yet. IMO, `set_group_weight` needs to be kept to enable changing the split-ratio between collator rewards and the treasury. 

For now, the rule is 3% annual inflation as block rewards of which Collators receive 1000 CFG per month.

Reply:
Ok, I understand 🙌🏻 

Only as a note, modifying the weight of a group does not have any effect if there is only one group. The weight is a proportion among groups. `effective_rate = weight_i/weight_total`

Reply:
Thanks for pointing this out, I need to rethink how to handle the split rewarding then without instantiating a group representing the Treasury 🥲